Onvif: Clarified the need to have at least one H.264 video stream present (#41570)
home-assistant/home-assistant.github.io: source/_integrations/onvif.markdown

### Configuration notes

> Most of the ONVIF devices support more than one audio/video profile. Each profile provides different image quality, or in the case of an NVR, separate connected cameras. This integration will add entities for all compatible profiles with the video encoding set to H.264, and only H.264, alternative formats like H.265 and MJPEG aren't supported so at least one video stream from each camera must be set to H.264 in order for the integration to find it. Usually, the first profile has the highest quality and it is the profile used by default. However, you may want to use a lower quality image. You may disable unwanted entities through the Home Assistant UI.

Reviewer: The original text already states that it must be set to H.264. It doesn't say anything else is supported.

Author: Yeah but it doesn't also clearly say that nothing else is supported.  There are a number of threads in different places where people are having problems with the integration apparently not working until they figure out, either by trial and error or because someone else who's run into the problem lets them know, that they need to enable H.264 somewhere for the integration to work.  It's the sort of thing that's obvious in hindsight but not obvious when you're sitting in front of an error message telling you that no usable cameras were found.

I'm not fussed about the wording (the suggestion is actually a bit clunky and could do with a cleanup), but there should be some text that explicitly tells users that things won't work unless there's at least one H.264 stream present.  Alternatively, change the error message to add something like "Have you remembered to enable an H.264 video stream on your camera?".

Reviewer: Hm, I see. Would it make sense to add a Troubleshooting section, describing the symptom and resolution? so that people who search for the error, like "No usable cameras were found," land there?

Author: Yup, that would work fine too.

Reviewer 2: @fencepost-error Could you make that adjustment?

Author: Sure.  How does the following sound?

### Troubleshooting

Many newer cameras, particularly ones with higher resolutions that benefit from H.265's improved video coding, support H.265 (HEVC) by default while the ONVIF integration looks for H.264 (AVC) video streams in order to find cameras.  If you get an error message "No usable cameras were found" then you need to update the camera configuration to output at least one video stream in H.264 format rather than H.265.  One option for doing this is to set a secondary stream to H.264 while leaving the primary stream at the default H.265.

Also, should I update this PR or close it out and submit a new PR for the above text?

Reviewer 2: You need to adjust the PR and mark it ready for review once you've processed the review comments. At that point, a new review would take place.

../Frenck

Author: OK, I've backed out the original changes and added the new text as above.

| IR lamp  | `ir_lamp` |  Turn infrared lamp on and off via `IrCutFilter` ONVIF imaging setting. |
| Autofocus  | `autofocus` |  Turn autofocus on and off via `AutoFocusMode` ONVIF imaging setting. |
| Wiper  | `wiper` |  Turn on the lens wiper on and off via the `Wiper` ONVIF auxiliary command. |

## Troubleshooting

### Error message: "No usable cameras were found"

> #### Symptom

Reviewer: ```suggestion
### Symptom: Error message: "No usable cameras were found"
```

| IR lamp  | `ir_lamp` |  Turn infrared lamp on and off via `IrCutFilter` ONVIF imaging setting. |
| Autofocus  | `autofocus` |  Turn autofocus on and off via `AutoFocusMode` ONVIF imaging setting. |
| Wiper  | `wiper` |  Turn on the lens wiper on and off via the `Wiper` ONVIF auxiliary command. |

> ### Troubleshooting

Reviewer: ```suggestion
## Troubleshooting

### Error message: "No usable cameras were found"
```

> Many newer cameras, particularly ones with higher resolutions that benefit from H.265's improved video coding, support H.265 (HEVC) by default while the ONVIF integration looks for H.264 (AVC) video streams in order to find cameras. If you get an error message "No usable cameras were found" then you need to update the camera configuration to output at least one video stream in H.264 format rather than H.265. One option for doing this is to set a secondary stream to H.264 while leaving the primary stream at the default H.265.

Reviewer: ```suggestion
Many newer cameras, particularly those with higher resolutions that benefit from H.265's improved video coding, support H.265 (HEVC) by default, while the ONVIF integration looks for H.264 (AVC) video streams in order to find cameras. If you get an error message "No usable cameras were found", then you need to update the camera configuration to output at least one video stream in H.264 format rather than H.265. One option for doing this is to set a secondary stream to H.264 while leaving the primary stream at the default H.265.
```

> Many newer cameras, particularly those with higher resolutions that benefit from H.265's improved video coding, support H.265 (HEVC) by default, while the ONVIF integration looks for H.264 (AVC) video streams in order to find cameras. If you get an error message "No usable cameras were found", then you need to update the camera configuration to output at least one video stream in H.264 format rather than H.265. One option for doing this is to set a secondary stream to H.264 while leaving the primary stream at the default H.265.

Reviewer: ```suggestion

#### Symptom

The ONVIF integration shows an error message "No usable cameras were found". 

#### Resolution

Update the camera configuration to output at least one video stream in H.264 format rather than H.265. One option for doing this is to set a secondary stream to H.264 while leaving the primary stream at the default H.265.

#### Cause 

Many newer cameras, particularly those with higher resolutions that benefit from H.265's improved video coding, support H.265 (HEVC) by default, while the ONVIF integration looks for H.264 (AVC) video streams to find cameras. 
```

Author: Looks fine!
